src/tests/unittests/hextobin.c

```diff
@@ -0,0 +1,123 @@
+/*
+ * hextobin.c: Test suite for sc_hex_to_bin()
+ *
+ * Copyright (C) 2022 Peter Popovec <popovec.peter@gmail.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdio.h>
+#include <stdint.h>
+#include <string.h>
+#include "libopensc/opensc.h"
+
+#define LEN 15
+
+int main()
+{
+	char *t;
+	char test[][LEN] = {
+/* result length (0x60 error string, 0x70 end of array), string, result string */
+
+		{1}, "0", "\x00",
+		{1}, ":0", "\x00",
+		{1}, " :0 :", "\x00",
+		{1}, "00", "\x00",
+		{1}, ":00", "\x00",
+		{1}, "0:0 :", "\x00",
+
+		{1}, "1", "\x01",
+		{1}, " 9", "\x09",
+		{1}, "d", "\x0d",
+		{1}, ":a", "\x0a",
+		{1}, " :b :", "\x0b",
+		{1}, " :c ", "\x0c",
+
+		{1}, "01", "\x01",
+		{1}, " 09", "\x09",
+		{1}, ":0a", "\x0a",
+		{1}, " :0b :", "\x0b",
+
+		{1}, "10", "\x10",
+		{1}, " 90", "\x90",
+		{1}, ":a0", "\xa0",
+		{1}, " :B0 :", "\xb0",
+
+		{1}, " 11:", "\x11",
+		{1}, " :11:", "\x11",
+		{1}, " :1 1:", "\x11",
+		{1}, "1:1", "\x11",
+		{1}, ":a1", "\xa1",
+		{1}, " :b:2 :", "\xb2",
+
+		{2}, " 12ab", "\x12\xab",
+		{2}, " :b:211 :", "\xb2\x11",
+		{2}, " ::1:2:a:b", "\x12\xab",
+
+		{3}, "b:211 :22", "\xb2\x11\x22",
+		{3}, "b2:11 :22", "\xb2\x11\x22",
+		{3}, "b2 11 22", "\xb2\x11\x22",
+		{3}, "1:234:56", "\x12\x34\x56",
+
+		{4}, "12:34:56:78", "\x12\x34\x56\x78",
+
+		{0}, "::::", "",
+		{0}, ":", "",
+		{0}, " ", "",
+		{0}, "", "",
+
+		/* incorrect strings */
+		{0x60}, "111", "",
+		{0x60}, "b:211 :2", "",
+		{0x60}, "G", "",
+		{0x60}, " z", "",
+		{0x60}, "1:11", "",
```

Comment:
I am wondering whats on `1:11` more wrong than on `1:234:56`, which is supposed to be a valid input.